src/frontend/src/catalog.rs

```diff
@@ -106,9 +124,91 @@ impl CatalogManager for FrontendCatalogManager {
 
     async fn register_system_table(
         &self,
-        _request: RegisterSystemTableRequest,
+        request: RegisterSystemTableRequest,
     ) -> catalog::error::Result<()> {
-        unimplemented!()
+        if let Some(dist_instance) = &self.dist_instance {
+            let open_hook = request.open_hook;
+            let request = request.create_table_request;
+
+            if self
+                .table(
+                    &request.catalog_name,
+                    &request.schema_name,
+                    &request.table_name,
+                )
+                .await?
+                .is_some()
+            {
+                return Ok(());
+            }
+
+            let time_index = request
```

Comment:
What will happen if the following code is executed concurrently

Reply:
The safety of concurrently creating tables is ensured by "create table procedure" which I am currently doing.